reference/constraints/Timezone.rst

```diff
@@ -85,8 +86,33 @@ string meant to contain a timezone identifier (e.g. ``America/New_York``):
 Options
 -------
 
+countryCode
+~~~~~~~~~~~
+
+**type**: ``string`` **default**: ``null``
+
+If the ``zone`` option is set to ``\DateTimeZone::PER_COUNTRY``, this option
+restricts the valid timezone identifiers to the ones that belong to the given
+country.
+
+The value of this option must be a valid `ISO 3166-1 alpha-2`_ country code
+(e.g. ``CN`` for China).
+
 .. include:: /reference/constraints/_groups-option.rst.inc
 
+intlCompatible
+~~~~~~~~~~~~~~
+
+**type**: ``boolean`` **default**: depends on your application
+
+If ``false``, only the `PHP timezone identifiers`_ are considered valid. If
+``true``, the :ref:`ICU timezones <component-inlt-timezones>` provided by the
+Intl component are considered valid too.
+
+The default value of this option depends on your application. If the
+:doc:`Intl component </components/intl>` is available in the application, the
+default value is ``true``; otherwise, it's ``false`` by default.
+
```

Comment:
so this option is really confusing i guess :)

both PHP+SF intl timezones are considered valid, regardless of this option. The SF intl timezones are only considered valid if the component is available.

However, because of 2 possible different versions of ICU (the version used to compile SF intl vs. the version used by ext-intl compiled in PHP) an edge case might occur:

if a timezone is considered valid by SF intl (based on the latest ICU version in SF), it might be unknown when using it in PHP (e.g. `$model = \IntlTimeZone::createTimeZone($tzValidatedBySfIntl);`) due a different ICU version PHP is compiled with.

Reply:
so the TLDR of this option is, it ensures you can use the timezone ID with `IntlTimeZone::createTimeZone($tz)` :) and you'll never get `Etc/Unknown`

Reply:
Yes, the name of this option is super confusing. The name tells me that true/false enables/diasbles support for Intl timezones.

But you say that Intl is unconditionally enabled. The, when should I set this option to true/false and what does it do? Thanks.

Reply:
did you see the last comment?

"intl" can mean 2 things: SF Intl or PHP Intl (https://www.php.net/manual/en/book.intl.php). Here it means the latter

We have no control of the ICU version used by PHP, hence this option to ensure a valid timezone ID when using `\IntlTimeZone` for a model (this class is provided by PHP Intl)

Reply:
so TLDR :D 

set this option to true if you're modelling the validated timezone ID with `\IntlTimeZone`, or aim to do so in the future and you want to prepare your data already

Reply:
Phewwww ... I think I finally understood this! Please review the latest reword ... and thanks for your patience 😅